content/blog/how-to-test-react-app-with-node-test-runner.en.mdx

```diff
@@ -0,0 +1,164 @@
+        Increment
+      </button>
+    </div>
+  );
+};
+
+export default Counter;
+```
+
+<details>
+  <summary>`index.module.css`</summary>
+
+  ```css fileName="src/components/Counter/index.module.css"
+  .container {
+    @apply flex flex-col items-center justify-center;
+
+    .count {
+      @apply text-4xl;
+    }
+
+    .button {
+      @apply px-4 py-2 bg-blue-500 text-white rounded-md;
+    }
+  }
+  ```
+</details>
+
+## Registering Node.js Loaders
+
+To handle TypeScript and CSS modules, you need to register the appropriate loaders. Create a file named `node-hooks/react-test.js` and add the following code:
+
+> If you want to understand what is a loader. You need to read [this post](/blog/post/how-to-use-nodejs-loader) I had written about loaders.
+
+```mjs fileName="node-hooks/react-test.js"
+import { register } from 'node:module';
+import { pathToFileURL } from 'node:url';
+import jsdom from 'jsdom';
+
+// Register the loaders
+register('nodejs-loaders/dev/tsx', pathToFileURL('./'));
+register('nodejs-loaders/testing/css-module', pathToFileURL('./'));
+
+// We don't need to load any HTML/CSS because it's handled by testing-library
+const { document } = new jsdom.JSDOM().window;
+
+/**
+ * Make the JSDOM global objects available to the Node.js environment.
+ */
+global.document = document;
+global.window = document.defaultView;
+```
+
+## Writing the Test
+
+Now, let's write a test for the `Counter` component. Create a file named `index.test.tsx` in the same directory as your component:
+
+```tsx fileName="src/components/Counter/index.test.tsx"
+import assert from 'node:assert/strict';
+import { describe, it } from 'node:test';
+import { render, fireEvent, screen } from '@testing-library/react';
+import Counter from './index.ts'; // ⚠️ We need to import the file with the .ts extension
+
+describe('Counter', () => {
+  it('should increment the count when the button is clicked', () => {
+    const { unmount } = render(<Counter />);
+
+    const button = screen.getByRole('button', { name: /increment/i });
+    const count = screen.getByText('0');
+
+    assert.strictEqual(count.textContent, '0');
+
+    fireEvent.click(button);
+
+    assert.strictEqual(count.textContent, '1');
+
+    // ⚠️ You must unmount the component instead the next test will still have the previous component in the "dom"
+    unmount();
+  });
+});
+```
+
+### Structure of a Test File
+
+A typical test file structure includes:
+
+1. **Imports**: Import the necessary modules and components.
+2. **Test Suite**: Define a test suite using `describe`.
+3. **Test Case**: Define individual test cases using `it`.
+4. **Render the Component**: Render the component to be tested.
+5. **Perform Actions**: Simulate user interactions or other actions.
+6. **Assertions**: Make assertions to verify the expected behavior.
+7. **Unmount the Component**: Clean up by unmounting the component.
+
+## Running the Test
+
+To run the test, use the following command:
+
+```bash
```

Comment:
`.env` file?

Reply:
Hum ?

Reply:
so put `NODE_OPTIONS` in `.env` ? I'm having a sub-paragraph made that says “if you have too large an command”.

Reply:
Also there is no good resource yet at link for .env support